test/extended/util/db/mongodb.go

```diff
@@ -0,0 +1,65 @@
+package db
+
+import (
+	"fmt"
+	"github.com/openshift/origin/test/extended/util"
+)
+
+// MongoDB is a MongoDB helper for executing commands.
+type MongoDB struct {
+	PodName       string
+	MasterPodName string
+}
+
+// NewMongoDB queries OpenShift for a pod with given name, saving environment
+// variables like username and password for easier use.
+func NewMongoDB(podName, masterPodName string) util.Database {
+	if masterPodName == "" {
+		masterPodName = podName
+	}
+	return &MongoDB{
+		PodName:       podName,
+		MasterPodName: masterPodName,
+	}
+}
+
+func (m MongoDB) GetPodName() string {
+	return m.PodName
+}
+
+func (m MongoDB) IsReady(oc *util.CLI) (bool, error) {
+	return isReady(
+		oc,
+		m.PodName,
+		"mongo --quiet --eval '{\"ping\", 1}'", "1",
+	)
+}
+
+func (m MongoDB) Query(oc *util.CLI, query string) (string, error) {
+	conf, err := getPodConfig(oc.KubeREST().Pods(oc.Namespace()), m.PodName)
+	if err != nil {
+		return "", err
+	}
+
+	command := fmt.Sprintf(
+		"echo '%s' | mongo --quiet '%s' --username '%s' --password '%s'",
+		query,
+		conf.Env["MONGODB_DATABASE"],
+		conf.Env["MONGODB_USER"],
+		conf.Env["MONGODB_PASSWORD"],
+	)
+
+	return executeShellCommand(oc, conf.Container, m.PodName, command)
+}
+
+// Query queries the database as a privileged user
+func (m MongoDB) QueryPrivileged(oc *util.CLI, query string) (string, error) {
+	// FIXME: not implemented
```

Comment:
Hm.. I added them because it doesn't compile without:

```
# github.com/openshift/origin/test/extended/util/db
_output/local/go/src/github.com/openshift/origin/test/extended/util/db/mongodb.go:23: cannot use MongoDB literal (type *MongoDB) as type "github.com/openshift/origin/test/extended/util".Database in return argument:
	*MongoDB does not implement "github.com/openshift/origin/test/extended/util".Database (missing QueryPrivileged method)
```